### PR TITLE
Albums

### PR DESCRIPTION
**Pull request by oleglukyanov**
Added a very basic album functionality. 
Albums can be created by the user and their names can be viewed at the albums page.

### CODE REVIEW

File: app/controllers/albums_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  def new
+    @user = current_user
+    @album = current_user.albums.new
+  end
+
+  def create
+    album = Album.new(album_params)
+    current_user.albums << album
+    current_user.save
+    redirect_to
```
Comment (dearshrewdwit) [inline app/controllers/albums_controller.rb]: redirect to where?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -23,6 +23,10 @@
   <body>
     <%= link_to 'Sign out', destroy_user_session_path, :method=>'delete' %>
 
+    <% if current_user %>
+      <%= link_to 'My albums', ('/users/' + current_user.id.to_s + '/albums') %>
```
Comment (dearshrewdwit) [inline app/views/layouts/application.html.erb]: you should be able to user a rails path helper here `users_albums_path(user)` something like this

File: config/routes.rb [hunk 3]
```diff
@@ -5,7 +5,11 @@
   # root 'posts'
 
   devise_for :users
+
   resources :posts
+  resources :users do
+    resources :albums
```
Comment (dearshrewdwit) [inline config/routes.rb]: nice

File: spec/factories/albums.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
```
Comment (dearshrewdwit) [inline spec/factories/albums.rb:1]: what's the need for this factory?

**Review by dearshrewdwit — COMMENTED** [review]
Please include a feature test for the `AlbumsController#index` action

File: app/views/albums/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<h1>My albums</h1>
+<br/>
+<a href="/users/<%= current_user.id %>/albums/new"><button>New album</button></a>
```
Comment (dearshrewdwit) [inline app/views/albums/index.html.erb]: this should be a rails path helper

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -24,6 +24,7 @@
     <%= link_to 'Sign out', destroy_user_session_path, :method=>'delete' %>
 
     <% if current_user %>
+      <%= link_to 'My albums', ('/users/' + current_user.id.to_s + '/albums') %>
```
Comment (dearshrewdwit) [inline app/views/layouts/application.html.erb]: this should be a rails path helper

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -24,6 +24,7 @@
...
       <a href="/users/<%= controller.current_user.id %>">My profile</a>
```
Comment (dearshrewdwit) [inline app/views/layouts/application.html.erb]: this should be a rails path helper

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

**Comment by oleglukyanov** [comment]
@dearshrewdwit Hi Ed, I've addressed the comments, can it be merged now?

File: spec/features/albums_can_be_created_and_viewed_spec.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  before do
+    visit("/users/sign_in")
+    fill_in 'user_email', with: user.email
+    fill_in 'user_password', with: user.password
+    within(:css, 'div.actions') do
+      click_button 'Log in'
+    end
+  end
+
+  scenario 'View albums' do
```
Comment (dearshrewdwit) [inline spec/features/albums_can_be_created_and_viewed_spec.rb:16]: `scenario 'View albums'`

this is testing that there are two links on the page, which is good but a different test.

for this test, i'd suggest to use the factory to create an album and then assert that its visibile to the user

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: spec/features/albums_can_be_created_and_viewed_spec.rb (hunk 8 above)
Comment (oleglukyanov) [reply spec/features/albums_can_be_created_and_viewed_spec.rb:16]: Updated (not sure if I get it right tho)

**Comment by oleglukyanov** [comment]
@dearshrewdwit Updated!

File: spec/features/albums_can_be_created_and_viewed_spec.rb (hunk 8 above)
Comment (dearshrewdwit) [reply spec/features/albums_can_be_created_and_viewed_spec.rb:16]: Looks great 🆗 

probably would be helpful to have those test that the user can use those links to navigate to the new album and my albums routes.

**Review by dearshrewdwit — APPROVED** [review]

**Pull request merged**